### PR TITLE
Equip float domain to handle definite NaN, +Inf, -Inf, and some missing float stubs

### PR DESCRIPTION
**Pull request by michael-schwarz**
In the course of #868 I came across a few tests for sv-comp that are low-hanging fruit that require handling the information that some variable is definitely NaN, +Inf or -Inf. This adds these possibilities.
The resulting lattice is a flat lattice with NaN, +Inf, -Inf, and intervals of actual floats values.

In the course of this PR, I will also add the float stubs sv-comp uses that I identified in #868.

### CODE REVIEW

File: tests/regression/57-floats/15-more-library.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    double c = ceil(-0.0);
+    __goblint_check((c == -0.0) && signbit(c)); //TODO (We do not distinguish +0 and -0)
+
+    c = ceil(-INFINITY);
+
+    // On OS X this gets expanded differently than on Linux where it is equivalent to the one below
+    // Might make sense to check what is needed for OS X support in the future, but this is not a deal-breaker
+    // and not high priority for now.
+    // __goblint_check(isinf(INFINITY) && signbit(c));
```
Comment (sim642) [inline tests/regression/57-floats/15-more-library.c:42]: So this (and the one below and in 57/17) could just be `// TODO` or does that cause bigger problems than just one check not passing?

**Review by sim642 — COMMENTED** [review]
I suppose the definite NaN/+inf/-inf domain could be extracted out to a separate float domain and be another component of `FloatDomTuple`, which could decouple things somewhat. But since it's already done, there's no point in redoing it right now.

File: tests/regression/57-floats/15-more-library.c (hunk 1 above)
Comment (michael-schwarz) [reply tests/regression/57-floats/15-more-library.c:42]: Good point!

File: src/analyses/libraryFunctions.ml [hunk 2]
```diff
@@ -145,21 +169,57 @@ let zstd_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     Functions and builtin versions of function and macros defined in math.h. *)
 let math_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     ("__builtin_nan", special [__ "str" []] @@ fun str -> Math { fun_args = (Nan (FDouble, str)) });
+    ("nan", special [__ "str" []] @@ fun str -> Math { fun_args = (Nan (FDouble, str)) });
```
[the remark this answers is not in the document]
Comment (michael-schwarz) [reply src/analyses/libraryFunctions.ml:172]: Yes, I added it now.

**Review by sim642 — APPROVED** [review]

**Pull request merged**